### ISSUE TITLE
New 

### ISSUE THREAD

**Issue by brandi83** [freeCodeCamp/freeCodeCamp#12971]
<<?xml version="1.0" encoding="utf-8"?>
<feed xmlns="http://www.w3.org/2005/Atom" xml:lang="en-us"><title>Code Review - All issues</title><link href="https://codereview.chromium.org/" rel="alternate"></link><link href="https://codereview.chromium.org/rss/all" rel="self"></link><id>https://codereview.chromium.org/</id><updated>2017-01-30T04:22:33+00:00</updated><author><name>rietveld</name></author><entry><title>[MD Bookmarks] Add Select for Bookmarks.
</title><link href="https://codereview.chromium.org/2639453002/" rel="alternate"></link><updated>2017-01-30T04:22:33+00:00</updated><author><name>jiaxi</name></author><id>urn:md5:a79dc58ec5e58687b666259d1ceeea31</id><summary type="html">
  [MD Bookmarks] Add Select for Bookmarks.

This CL enables single click to select items in the list.
- Shift select.
- Ctrl select.

BUG=658980
CQ_INCLUDE_TRYBOTS=master.tryserver.chromium.linux:closure_compilation

</summary></entry><entry><title>[async-functions] support await expressions in finally statements
</title><link href="https://codereview.chromium.org/2654423004/" rel="alternate"></link><updated>2017-01-30T04:20:29+00:00</updated><author><name>caitp</name></author><id>urn:md5:57f13564212e076be4623874c2fd879d</id><summary type="html">
  [async-functions] support await expressions in finally statements

BUG=v8:5896
R=gsathya@chromium.org, rmcilroy@chromium.org, neis@chromium.org

</summary></entry><entry><title>Refactor how oauth token cache is organized.
</title><link href="https://codereview.chromium.org/2659263002/" rel="alternate"></link><updated>2017-01-30T04:19:22+00:00</updated><author><name>Vadim Sh.</name></author><id>urn:md5:eb62e11ac877aa2adc2a8e326c0d8059</id><summary type="html">
  Refactor how oauth token cache is organized.

Also add more docs and tests.

The previous implementation had a number of shortcomings:
  * It used not-very-descriptive digest-based key for the cache file name, and
    didn&amp;#39;t write any details about the origin of the token into the cache file.
    As a result, the cache directory ends up looking like a mess with a bunch of
    randomly-looking files.
  * It was possible to have a collision in names of the cache entries.
  * There was no automatic cleanup of old entries.
  * Cache key for service account JSONs was partially derived from the location
    of the key file, not its content (key ID). It was a hack.

New implementation stores all cached tokens in a single nice and tidy JSON file.

This allows to use structured cache keys, so we no longer lose information about
cache entry origins. It is also makes easier to remove old entries from
the cache.

The downside is a possibility of some token update being lost if the cache file
is being updated concurrently. This is bad only if the lost update is a cache
entry with new refresh token. This can happen only if &amp;quot;login&amp;quot; operation races
with some other operation that updates the cache. &amp;quot;login&amp;quot; happens very-very
infrequently, and only on end-user machines.

On bots (that exclusively use service accounts), a lost write to the cache is
merely an inefficiency: next time the process starts, it will have to redo
OAuth2 flow (1 HTTP request) to get a new access token in place of the lost one.

Once this is deployed, it will effectively &amp;quot;logout&amp;quot; everyone. At some point
later (when we are sure no code that uses old cache structure is running), we&amp;#39;ll
add a change that cleans up old *.tok cache items.

R=iannucci@chromium.org
BUG=

</summary></entry><entry><title>Migrate external protocol prefs from local state to profiles
</title><link href="https://codereview.chromium.org/2623033006/" rel="alternate"></link><updated>2017-01-30T04:18:46+00:00</updated><author><name>ramyasharma</name></author><id>urn:md5:bb15c95f5a306a8392af3e84103ed683</id><summary type="html">
  Migrate external protocol prefs from local state to profiles.

Moves external protocol handler state from local state to per-profile.
This ensures that external protocol settings maybe distinct per-profile
and simplifies the implementation of editing UI.

The local state data is migrated to the currently active profile and then
deleted. This may result in some loss for people who use multiple
profiles; However, they will be presented with an external protocol
dialog the next time they try and open a formerly remembered scheme.

It also resets the state of all schemes which have been permanently
blocked by users (and are not part of the pre-populated set of schemes).
These schemes will prompt the user the next time they are accessed. We
do this because:
a) there is currently no UI to edit external handlers, so users cannot
recover from permanently silencing a dialog.
b) local state is per Chrome instance, meaning that users have to wipe
their entire Chrome data dir (or edit the local state JSON file) to recover.

BUG=457254,62799

</summary></entry><entry><title>Add UMA for autoblocking and embargoing.
</title><link href="https://codereview.chromium.org/2651163002/" rel="alternate"></link><updated>2017-01-30T04:18:46+00:00</updated><author><name>meredithl</name></author><id>urn:md5:f228ecd3b670d06110af2ec43b7a873e</id><summary type="html">
  Add UMA for autoblocking and embargoing.

Metrics are recorded in PermissionUmaUtil. Currently, the client records
what the response time was from Safe Browsing, as well as what the
response was. The autoblocker records embargo status for both
Permissions Blacklisting and repeated dismissals. Recording that a permission
was not embargoed happens in either PermissionsContextBase once the permission
has been responded to, or in PermissionRequestImpl when it is destroyed after
not being responded to. These are mutually exclusive events so they should not
interfere with one another.

BUG=679877

</summary></entry><entry><title>Add listener for AdapterDiscoveringChanged on BT chooser-controller
</title><link href="https://codereview.chromium.org/2663693002/" rel="alternate"></link><updated>2017-01-30T04:14:25+00:00</updated><author><name>deejay</name></author><id>urn:md5:8b804297ab5e9f92b9a4f719b01fcfca</id><summary type="html">
  Add listener for AdapterDiscoveringChanged on BT chooser-controller

|AdapterDiscoveringChanged| can be called by device/bluetooth
when BT discovery stopped.

We add listener for that on bluetooth chooser-controller.
Then it will stops the discovery session and notifies
the chooser when discovery stops accidentally.

BUG=611852

</summary></entry><entry><title>Roll src/third_party/skia/ 5ce33efa7..55b72530f (1 commit).
</title><link href="https://codereview.chromium.org/2663683002/" rel="alternate"></link><updated>2017-01-30T04:11:13+00:00</updated><author><name>skia-deps-roller</name></author><id>urn:md5:a841362075682e63b0ae3f6ed88c50eb</id><summary type="html">
  Roll src/third_party/skia/ 5ce33efa7..55b72530f (1 commit).

https://skia.googlesource.com/skia.git/+log/5ce33efa7c91..55b72530fede

$ git log 5ce33efa7..55b72530f --date=short --no-merges --format=&amp;#39;%ad %ae %s&amp;#39;
2017-01-29 benjaminwagner Omit new test from Bazel ASAN.

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

If the roll is causing failures, see:
http://www.chromium.org/developers/tree-sheriffs/sheriff-details-chromium#TOC-Failures-due-to-DEPS-rolls

CQ_INCLUDE_TRYBOTS=master.tryserver.blink:linux_trusty_blink_rel
TBR=halcanary@google.com

</summary></entry><entry><title>Changed EPosition to an enum class and renamed its members
</title><link href="https://codereview.chromium.org/2667493002/" rel="alternate"></link><updated>2017-01-30T04:09:10+00:00</updated><author><name>shend</name></author><id>urn:md5:869d8aa597a9bc7bc56a7aefdf5baff5</id><summary type="html">
  Changed EPosition to an enum class and renamed its members

Changed EPosition to an enum class and gave it an unsigned underlying
type. Also renamed its members to match the CSS keywords they
represent, and added the &amp;#39;k&amp;#39; prefix in line with the Blink rename to
match Chromium style.

Changing it to an enum class enforces better namespacing and code
practices. Adding the unsigned underlying type is pre-work for when the
class is eventually stored as an enum bitfield (it would be done in this
patch, except a presubmit warning already exists that prevents that. The
presubmit warning needs to be updated before that change can occur.)

This is also pre-work to allow EPosition to be used by generated
fields in ComputedStyleBase.

BUG=684966
CQ_INCLUDE_TRYBOTS=master.tryserver.chromium.linux:linux_layout_tests_slimming_paint_v2

</summary></entry><entry><title>Add picker view for Web Share Target.
</title><link href="https://codereview.chromium.org/2644793006/" rel="alternate"></link><updated>2017-01-30T04:07:13+00:00</updated><author><name>Matt Giuca</name></author><id>urn:md5:f53fdc72302e4db44ec22f8955f47d99</id><summary type="html">
  Add picker view for Web Share Target.

This is temporary UI which has not been approved by UX and will only be
used for an experimental feature behind a flag.

Hooked up to the share service for navigator.share on Desktop. Currently
does not show any real apps (just the dummy app which is currently
hard-coded in the share service).

BUG=668389

</summary></entry><entry><title>Call Element::rebuildLayoutTree from Document::updateStyle directly
</title><link href="https://codereview.chromium.org/2473743003/" rel="alternate"></link><updated>2017-01-30T04:05:12+00:00</updated><author><name>nainar</name></author><id>urn:md5:0c63b7e3cfee7cb7a0f2ce0ea139edcc</id><summary type="html">
  Call Element::rebuildLayoutTree from Document::updateStyle directly

This patch changes the callstack in Document::updateStyle to call
Element::rebuildLayoutTree after Element::recalcStyle instead of
from it.

It also adds helper functions needed to perform Layout Tree Construction:
1. ContainerNode::rebuildDescendantLayoutTree()
2. Element::rebuildShadowRootLayoutTree()
3. Element::reattachPseudoElementLayoutTree(PseudoId pseudoId)
4. ShadowRoot::rebuildLayoutTree()

It also changes the setting/getting of the (Child)NeedsStyleRecalc
and (Child)NeedsReattachLayoutTree bits to match this pattern.

BUG=595137, 658092

</summary></entry><entry><title>Redo &amp;quot;dart2js: Redo known-type analysis after load-elimination&amp;quot;
</title><link href="https://codereview.chromium.org/2662493004/" rel="alternate"></link><updated>2017-01-30T04:04:08+00:00</updated><author><name>sra1</name></author><id>urn:md5:2ea2ad356b955ceb87a022fa2f0d29f9</id><summary type="html">
  Redo &amp;quot;dart2js: Redo known-type analysis after load-elimination&amp;quot;

BUG=

</summary></entry><entry><title>Add UseCounter for ServiceWorkerNavigationPreload.
</title><link href="https://codereview.chromium.org/2659043002/" rel="alternate"></link><updated>2017-01-30T04:02:51+00:00</updated><author><name>falken</name></author><id>urn:md5:9f982366a657999ea7c87f505259ca27</id><summary type="html">
  Add UseCounter for ServiceWorkerNavigationPreload.

We have a UMA already (ServiceWorker.NavigationPreload.HeaderSize) which gives
the total count, but it&amp;#39;s hard to compute the % of page views without a
UseCounter.

BUG=649558

</summary></entry><entry><title>Roll chromium_revision b84d9d8be2..169ed39de4 (446949:446956)
</title><link href="https://codereview.chromium.org/2667523002/" rel="alternate"></link><updated>2017-01-30T03:52:38+00:00</updated><author><name>buildbot</name></author><id>urn:md5:36ebba63ac4a49b42c80bbd593e6634a</id><summary type="html">
  Roll chromium_revision b84d9d8be2..169ed39de4 (446949:446956)

Change log: https://chromium.googlesource.com/chromium/src/+log/b84d9d8be2..169ed39de4
Full diff: https://chromium.googlesource.com/chromium/src/+/b84d9d8be2..169ed39de4

Changed dependencies:
* src/base: https://chromium.googlesource.com/chromium/src/base/+log/734d7cdd3e..d950673275
* src/third_party: https://chromium.googlesource.com/chromium/src/third_party/+log/b289529604..126aa8b373
* src/tools: https://chromium.googlesource.com/chromium/src/tools/+log/15417ab314..bb4947c144
DEPS diff: https://chromium.googlesource.com/chromium/src/+/b84d9d8be2..169ed39de4/DEPS

No update to Clang.

TBR=
BUG=None

</summary></entry><entry><title>Add WidgetTest::GetAllWidgets() to find dialogs created by TestBrowserDialog.
</title><link href="https://codereview.chromium.org/2660813002/" rel="alternate"></link><updated>2017-01-30T03:49:29+00:00</updated><author><name>tapted</name></author><id>urn:md5:68d470522d128f68c855e498aab246e4</id><summary type="html">
  Add WidgetTest::GetAllWidgets() to find dialogs created by TestBrowserDialog.

TestBrowserDialog currently fails to detect some browser dialogs on
Windows 10.

This is because TestBrowserDialog uses the set difference from calls to
Widget::GetAllChildWidgets() to detect when a dialog is added to a
browser window. On Aura, this is populated from Aura child windows (in
the same WindowTreeHost). However, a parented Widget using a
DesktopNativeWidgetAura (e.g. for a top-level Widget on Windows Aero),
rather than a NativeWidgetAura, will not be reported in its parent
window&amp;#39;s &amp;quot;GetAllChildWidgets()&amp;quot; (nor in Widget::GetAllOwnedWidgets()).
Ownership of these top-level Widgets is managed in mus-, win-, or
x11-specific code.

To fix, add WidgetTest::GetAllWidgets() which just returns all Widget*s
that can be found in the current process.

BUG=683808, 654151

</summary></entry><entry><title>DevTools: Regex checkbox was missing from filter bars
</title><link href="https://codereview.chromium.org/2659303002/" rel="alternate"></link><updated>2017-01-30T03:44:37+00:00</updated><author><name>einbinder</name></author><id>urn:md5:2397243be6a7b38927f5bc583a0aef8a</id><summary type="html">
  DevTools: Regex checkbox was missing from filter bars

BUG=686365

</summary></entry><entry><title>DevTools: Missing parenthesis on url in NetworkLog
</title><link href="https://codereview.chromium.org/2667513002/" rel="alternate"></link><updated>2017-01-30T03:44:36+00:00</updated><author><name>einbinder</name></author><id>urn:md5:c3b6eb2d50e6dd6766ceb6ac168be3bc</id><summary type="html">
  DevTools: Missing parenthesis on url in NetworkLog

BUG=none

</summary></entry><entry><title>Regenerate MANIFEST.json when WPT tests are run
</title><link href="https://codereview.chromium.org/2644783003/" rel="alternate"></link><updated>2017-01-30T03:41:49+00:00</updated><author><name>jeffcarp</name></author><id>urn:md5:91a074ec403a1564f86e1222e8546c4e</id><summary type="html">
  Regenerate MANIFEST.json when WPT tests are run

BUG=666957
R=qyearsley@chromium.org

</summary></entry><entry><title>Changed ETableLayout to an enum class and renamed its members
</title><link href="https://codereview.chromium.org/2666483002/" rel="alternate"></link><updated>2017-01-30T03:41:36+00:00</updated><author><name>shend</name></author><id>urn:md5:3cf877306af8acaaf7b091f153b71a4e</id><summary type="html">
  Changed ETableLayout to an enum class and renamed its members

Changed ETableLayout to an enum class and gave it an unsigned underlying
type. Also renamed its members to match the CSS keywords they
represent, and added the &amp;#39;k&amp;#39; prefix in line with the Blink rename to
match Chromium style.

Changing it to an enum class enforces better namespacing and code
practices. Adding the unsigned underlying type is pre-work for when the
class is eventually stored as an enum bitfield (it would be done in this
patch, except a presubmit warning already exists that prevents that. The
presubmit warning needs to be updated before that change can occur.)

This is also pre-work to allow ETableLayout to be used by generated
fields in ComputedStyleBase.

BUG=684966

</summary></entry><entry><title>Clean up ARC file system unit tests.
</title><link href="https://codereview.chromium.org/2651883003/" rel="alternate"></link><updated>2017-01-30T03:39:36+00:00</updated><author><name>Shuhei Takahashi</name></author><id>urn:md5:e91520686b498c2d97c72c6208e7d108</id><summary type="html">
  Clean up ARC file system unit tests.

1. Introduce a generic FakeFileSystemInstance.

Tests that want to test ARC file system operations can use this fake without
implementing its own FileSystemInstance.

2. Removed ArcFileSystemOperationRunner interface.

Former ArcDeferredFileSystemOperationRunner is now the only implementation
of the interface, so the interface is removed and ArcDeferredFSOpRunner is
renamed to ArcFSOpRunner.

3. Inject FileSystemInstance rather than ArcFSOpRunner in unit tests.

In unit tests, ArcFSOpRunner without deferring is created with
ArcFSOpRunner::CreateForTesting() so that operations are passed through
to FileSystemInstance.

TEST=unit_tests
BUG=chromium:683049

</summary></entry><entry><title>Read share targets from prefstore, and filter by engagement.
</title><link href="https://codereview.chromium.org/2666493002/" rel="alternate"></link><updated>2017-01-30T03:36:12+00:00</updated><author><name>constantina</name></author><id>urn:md5:fad6c030677e1fd055252f3a9ea2c060</id><summary type="html">
  Read share targets from prefstore, and filter by engagement.

New function, GetTargetsWithSufficientEngagement, called when browser
receives Share from renderer. Function collates all share targets and
their templates, and keeps targets that have a high enough engagement
score with the user.

Then replaces placeholders in the templates, and removes targets that
fail replacement, or result in invalid URLs.

The &amp;quot;picked&amp;quot; share target, is the first in the list after filtering.
This will be replaced in a follow up CL by the actual target picked by a
user via the UI in CL /2644793006.

BUG=668389

</summary></entry></feed>-- FreeCodeCamp Issue Template -->

#### Browser Information

* Browser Name, Version: 
* Operating System: 
* Mobile, Desktop, or Tablet: 

#### Your Code

```js
 
 
 
```
